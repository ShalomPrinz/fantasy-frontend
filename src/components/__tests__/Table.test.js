import React from 'react';
import renderer from 'react-test-renderer';
import { Table } from '../';

let tree = null;

beforeEach(() => {
    tree = null
})

const getJSON = (Component) => renderer.create(Component).toJSON()

describe('render table without data', () => {
    it('should not render anything', () => {
        tree = getJSON(<Table />);

        expect(tree).toMatchSnapshot();
    })
})

describe('render table with missing data', () => {
    it('should not render table at all', () => {
        const columns = [{ path: 'label' }]

        tree = getJSON(<Table columns={columns} />)

        expect(tree).toMatchSnapshot();
    })
    
    it('should not render any data when no columns supplied', () => {
        const data = [{ id: 1, label: 'Label' }]

        tree = getJSON(<Table data={data} />)

        expect(tree).toMatchSnapshot();
    })
    
    it('should render data also for rows without the specified data path', () => {
        const data = [{ id: 1, label: 'Label' }, { id: 2 }]
        const columns = [{ path: 'label' }]

        tree = getJSON(<Table data={data} columns={columns} />)
        
        expect(tree).toMatchSnapshot();
    })
})

describe('render table with full data', () => {
    it('should render a scrollable table', () => {
        const data = [{ id: 1, label: 'Label' }]
        const columns = [{ path: 'label'}]

        tree = getJSON(<Table data={data} columns={columns} scrollable />)

        expect(tree).toMatchSnapshot();
    })
    
    it('should render a styled table', () => {
        const data = [{ id: 1, label: 'Label' }]
        const columns = [{ path: 'label'}]

        tree = getJSON(<Table data={data} columns={columns} className="bg-default" />)

        expect(tree).toMatchSnapshot();
    })
})