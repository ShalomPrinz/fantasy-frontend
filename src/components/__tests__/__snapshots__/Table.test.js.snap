// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render table with full data should render a scrollable table 1`] = `
<div
  className="table-responsive mt-0 border-bottom scrollable"
>
  <table
    className="table rounded"
  >
    <tbody>
      <tr>
        <td
          className="align-middle"
        >
          Label
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`render table with full data should render a styled table 1`] = `
<div
  className="table-responsive mt-0"
>
  <table
    className="table rounded bg-default"
  >
    <tbody>
      <tr>
        <td
          className="align-middle"
        >
          Label
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`render table with missing data should not render any data when no columns supplied 1`] = `
<div
  className="table-responsive mt-0"
>
  <table
    className="table rounded"
  >
    <tbody>
      <tr />
    </tbody>
  </table>
</div>
`;

exports[`render table with missing data should not render table at all 1`] = `null`;

exports[`render table with missing data should render data also for rows without the specified data path 1`] = `
<div
  className="table-responsive mt-0"
>
  <table
    className="table rounded"
  >
    <tbody>
      <tr>
        <td
          className="align-middle"
        >
          Label
        </td>
      </tr>
      <tr>
        <td
          className="align-middle"
        />
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`render table without data should not render anything 1`] = `null`;
